### PR TITLE
Documentation for contributor guide #5199

### PR DESCRIPTION
**Pull request by Shekharcodess**
I have updated the documentation for the contributors guide and I have made sure all the mentioned points are covered

### CODE REVIEW

**Review by koba1t — CHANGES_REQUESTED** [review]
Hi @Shekharcodess

Thanks for your first contribution!
The code base of `kustomize` is just a bit complicated and doesn't use typical commands on `Makefile`.
Could you check for `Makefile` on the repo root and try that make commands on your computer?

And I recommend checking the below two files for what is running on CI.
- https://github.com/kubernetes-sigs/kustomize/blob/da4e8810079e2a98a3f4caad83618c53014d703b/.github/workflows/go.yml
- https://github.com/kubernetes/test-infra/blob/da4337756e8bba3a39a5cfb21997b2d35d1171f2/config/jobs/kubernetes-sigs/kustomize/kustomize-presubmit-master.yaml

**Comment by annasong20** [comment]
Hi @Shekharcodess, are you still working on this PR? We plan to increase its priority, so if you aren't someone else might assign #5199 to themselves.

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -31,6 +31,46 @@ Dev guides:
...
+`git commit -m "Description of your changes"`  
+`git push origin my-branch-name`  
+Visit the original Kustomize repository and click on the "Compare & pull request" button to create a PR with your changes.  
+
+
+
+Building Kustomize Locally:
+
+
+Ensure that you have Go installed on your system. Refer to the [Go](https://golang.org/doc/install) installation guide for your platform.
```
Comment (varshaprasad96) [inline CONTRIBUTING.md:54]: It would be nice to specify that we recommend to use the version supported in go mod.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -31,6 +31,46 @@ Dev guides:
...
+
+
+Building Kustomize Locally:
+
+
+Ensure that you have Go installed on your system. Refer to the [Go](https://golang.org/doc/install) installation guide for your platform.  
+Open a terminal and navigate to the root directory of your cloned Kustomize repository.  
+Run the following command to build Kustomize: `make build`  
+This will compile the Kustomize binary, and you can find it in the bin directory.  
+Optionally, you can add the bin directory to your system's PATH environment variable to use the locally built Kustomize binary.
```
Comment (varshaprasad96) [inline CONTRIBUTING.md:58]:
```suggestion
Optionally, you can add the bin directory to your system's $PATH environment variable to use the locally built Kustomize binary.  
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -31,6 +31,46 @@ Dev guides:
...
+Building Kustomize Locally:
+
+
+Ensure that you have Go installed on your system. Refer to the [Go](https://golang.org/doc/install) installation guide for your platform.  
+Open a terminal and navigate to the root directory of your cloned Kustomize repository.  
+Run the following command to build Kustomize: `make build`  
+This will compile the Kustomize binary, and you can find it in the bin directory.  
+Optionally, you can add the bin directory to your system's PATH environment variable to use the locally built Kustomize binary.  
+
+Overview of Important Make Targets and Running Tests Locally:
```
Comment (varshaprasad96) [inline CONTRIBUTING.md:60]: Nit:
```suggestion
Overview of important `Make` targets and running tests locally:
```

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -31,6 +31,46 @@ Dev guides:
...
+
+The Kustomize repository provides several make targets to perform different tasks. Here are some of the important ones:  
+`make test`: Run the tests for Kustomize locally.
```
[the remark this answers is not in the document]
Comment (varshaprasad96) [reply CONTRIBUTING.md:63]: Neither is there `test-e2e` `coverage`

**Comment by natasha41575** [comment]
Closing due to lack of activity

**Pull request closed without merging**